radix-engine/src/engine/id_allocator.rs

```diff
@@ -1,99 +1,107 @@
+use scrypto::rust::ops::Range;
 use scrypto::types::*;
 use scrypto::utils::*;
 
-pub const ECDSA_TOKEN_RID: Rid = Rid(0);
 pub const ECDSA_TOKEN_BID: Bid = Bid(0);
-const USER_RID_START: u32 = 1024;
-const USER_BID_START: u32 = 1024;
+pub const ECDSA_TOKEN_RID: Rid = Rid(1);
+
+pub const SYSTEM_ID_SPACE: IdSpace = IdSpace::System(0..512);
+pub const TRANSACTION_ID_SPACE: IdSpace = IdSpace::Transaction(512..1024);
+pub const APPLICATION_ID_SPACE: IdSpace = IdSpace::Application(1024..u32::MAX);
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum IdSpace {
+    System(Range<u32>),
+    Transaction(Range<u32>),
+    Application(Range<u32>),
```

Comment:
Doesnt seem right to have `Range` in the enums as it is exposed to caller. Rather, better to just match below on line 33 and set the range there.